### DISCUSSION TITLE
Show total time per day when binary sensor was "on"

### DISCUSSION THREAD

**Discussion by larius-z** [RomRider/apexcharts-card#260] [Q&A] [answered]
Hi,
I have a binary_sensor which is set to ON while the heater is ON. 
I would like to present a diagram showing how many hours per day the heater is ON.
Ideally it should present 7 or 14 days back, and additionally compare week to previous week.

Could you please help and suggest what should be the script to generate such diagram? 

I was looking for something simple which is not using utility meters etc. but just uses the native data of the binary sensor history.. Not sure if possible :)

Thanks for suggestions!

**Comment by RomRider** [comment, marked as answer]
You'll want to use this: https://www.home-assistant.io/integrations/history_stats

Then use those new entities in the chart.

**Comment by larius-z** [comment]
Sometimes you can miss the most obvious things :) Thanks, this solved my problem.

Btw. I already had history_stats & mini-grapth-card -> this setup is not sufficient to achieve what I mentioned in this topic. It requires also setting up utility_meter which seems to be redundant in this case..
